rasa/shared/core/generator.py

```diff
@@ -217,6 +217,40 @@ def update(self, event: Event, skip_states: bool = False) -> None:
 
             self._append_current_state()
 
+    def fingerprint(self) -> Text:
+        """Returns a unique hash for the tracker which is stable across python runs.
+
+        Returns:
+            fingerprint of the tracker
+        """
+        data = []
+        sender_id_hash = rasa.shared.utils.io.get_text_hash(self.sender_id)
+        data.append(sender_id_hash)
+
+        if self.domain:
+            data.append(self.domain.fingerprint())
+
+        if self.slots:
+            for slot in self.slots.values():
+                data.append(slot.fingerprint())
+
+        if self._states_for_hashing:
```

Comment:
I'm not an expert on this area of the code, but I wonder if we need to fingerprint this stuff if we just fingerprinted `self.events` instead (maybe in the base class)?

Reply:
Yup, removed this part and fingerprinted `self.events` instead.